controllers/authenticate.go

```diff
@@ -21,55 +21,44 @@ func NewAuthenticateController(getProfile func(string) (*auth0.Profile, error))
 	}
 }
 
-func (a *AuthenticateController) Routes() []*fireball.Route {
-	return []*fireball.Route{
-		{
-			Path: "/authenticate",
-			Handlers: fireball.Handlers{
-				"POST": a.authenticate,
-			},
-		},
-	}
-}
-
-func (a *AuthenticateController) authenticate(c *fireball.Context) (fireball.Response, error) {
-	defer c.Request.Body.Close()
-
-	var tr authentication.TokenReview
-	if err := json.NewDecoder(c.Request.Body).Decode(&tr); err != nil {
+func (a *AuthenticateController) Authenticate(r *http.Request) http.Handler {
```

Comment:
Just re-iterating previous comments about the need to call out how different this authentication API is  versus the UI-driven API